receiver/telemetrygeneratorreceiver/metrics_generator.go

```diff
@@ -0,0 +1,130 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package telemetrygeneratorreceiver //import "github.com/observiq/bindplane-agent/receiver/telemetrygeneratorreceiver"
+
+import (
+	"math"
+	"math/rand"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+// hostMetricsGenerator is a generator for host metrics. It generates a sampling of host metrics
+// emulating the Host Metrics receiver: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver
+type metricsGenerator struct {
+	otlpGenerator
+	// dataPointUpdaters is a list of functions that update the data points with new values
+	// for each batch of metrics that are generated
+	dataPointUpdaters []func()
+}
+
+func newMetricsGenerator(cfg GeneratorConfig, logger *zap.Logger) metricGenerator {
+
+	g := &metricsGenerator{
+		otlpGenerator: otlpGenerator{
+			cfg:     cfg,
+			logger:  logger,
+			logs:    plog.NewLogs(),
+			metrics: pmetric.NewMetrics(),
+			traces:  ptrace.NewTraces(),
+		},
+	}
+	// Load up Resources attributes
+
+	newResource := g.metrics.ResourceMetrics().AppendEmpty()
+	err := newResource.Resource().Attributes().FromRaw(cfg.ResourceAttributes)
+	if err != nil {
+		// should be caught in validation
+		logger.Error("Error setting resource attributes in host_metrics", zap.Error(err))
```

Comment:
If we do hit this error is it safe to continue?

Reply:
I think so, since it should be caught in validation, and if it isn't we'll omit the malformed attributes